engine/server/web_server/web_controller.hpp

```diff
@@ -334,6 +334,39 @@ class WebController : public oatpp::web::server::api::ApiController {
     return createDtoResponse(Status::CODE_200, status_dto);
   }
 
+  ADD_CORS(DeleteRecordsByPK)
+
+  ENDPOINT("POST", "/api/{db_name}/data/delete/pk", DeleteRecordsByPK,
```

Comment:
Can we just use ""/api/{db_name}/data/delete"? Later we can use whether primaryKeys exist and whether filter condition exists to distinguish which underneath function to call

Reply:
there's already `api/{db_name}/data/delete` path to delete by key. do you have desire payload format to differentiate different conditions?

Reply:
there's already api/{db_name}/data/delete path to delete by key. do you have desire payload format to differentiate different conditions?

Reply:
Actually that api/{db_name}/data/delete is exactly the path we should use. If you check the implementation, it has everything about semantic check on payload, except doing the actual deletion. We should make the implementation there

Reply:
ok, I thought that was used to delete by entry id. updated in https://github.com/epsilla-cloud/vectordb/pull/50/commits/329e4bde3fc5eaf0a00fe7b8b07a7b08a85b4d1c